script/validate-wasm-grammar-prs.js

```diff
@@ -0,0 +1,141 @@
+}
+
+const cmd = cp.spawnSync("git", [ "diff", "--name-only", "-r", "HEAD", commit])
+
+if (cmd.status !== 0 || cmd.stderr.toString().length > 0) {
+  console.error("Git Command has failed!");
+  console.error(`'git diff --name-only -r HEAD ${commit}'`);
+  console.error(cmd.stderr.toString());
+  process.exit(1);
+}
+
+const changedFiles = cmd.stdout.toString().split("\n");
+// This gives us an array of the name and path of every single changed file from the last two commits
+// Now to check if there's any changes we care about.
+
+if (verbose) {
+  console.log("Array of changed files between commits:");
+  console.log(changedFiles);
+}
+
+const wasmFilesChanged = changedFiles.filter(element => element.endsWith(".wasm"));
+
+if (wasmFilesChanged.length === 0) {
+  // No WASM files have been modified. Return success
+  console.log("No WASM files have been changed.");
+  process.exit(0);
+}
+
+// Now for every single wasm file that's been changed, we must validate those changes
+// are also accompanied by a change in the `parserSource` key
+
+for (const wasmFile of wasmFilesChanged) {
+  const wasmPath = path.dirname(wasmFile);
+
+  const files = fs.readdirSync(path.join(wasmPath, ".."));
+  console.log(`Detected changes to: ${wasmFile}`);
+
+  if (verbose) {
+    console.log("Verbose file check details:");
+    console.log(wasmFile);
+    console.log(wasmPath);
+    console.log(files);
+    console.log("\n");
+  }
+
+  for (const file of files) {
+    const filePath = path.join(wasmPath, "..", file);
+    console.log(`Checking: ${filePath}`);
+
+    if (fs.lstatSync(filePath).isFile()) {
+      const contents = CSON.readFileSync(filePath);
+
+      // We now have the contents of one of the grammar files for this specific grammar.
+      // Since each grammar may contain multiple grammar files, we need to ensure
+      // that this particular one is using the tree-sitter wasm file that was
+      // actually changed.
+      const grammarFile = contents.treeSitter?.grammar ?? "";
+
+      if (path.basename(grammarFile) === path.basename(wasmFile)) {
+        // This grammar uses the WASM file that's changed. So we must ensure our key has also changed
+        // Sidenote we use `basename` here, since the `wasmFile` will be
+        // a path relative from the root of the repo, meanwhile `grammarFile`
+        // will be relative from the file itself
+
+        // In order to check the previous state of what the key is, we first must retreive the file prior to this PR
+        const getPrevFile = cp.spawnSync("git", [ "show", `${commit}:./${filePath}` ]);
+
+        if (getPrevFile.status !== 0 || getPrevFile.stderr.toString().length > 0) {
+          // This can fail for two major reasons
+          // 1. It actually failed
+          // 2. This is a new file, and it failed to find an earlier one that didn't exist
+          // So that we don't faile brand new TreeSitter grammars, we manually check for number 2
```

Comment:
```suggestion
          // So that we don't fail brand new TreeSitter grammars, we manually check for number 2
```

Small typo in comment

Reply:
Also the wording of this comment over all wasn't very clear to me at first, but I figured it out after not much time.

> "It actually failed"

Means the `git show` command had an actual error status code

> "check for number 2"

Means check for reason 2

> "          // 2. This is a new file, and it failed to find an earlier one that didn't exist"

--> "failed to find an earlier copy (which didn't exist)"

Possibly rephrased like this?

But it's just a comment, and I understood it without too much difficulty, to be honest.